crates/iota-sdk/src/lib.rs

```diff
@@ -644,3 +636,87 @@ impl DataReader for ReadApi {
         Ok(self.get_reference_gas_price().await?)
     }
 }
+
+pub trait GetAllPages<O, C, F, E>: Sized + Fn(Option<C>) -> F
```

Comment:
Can you write a tiny bit of docs for the trait + methods please?

Reply:
Sure. I just wanted to get an initial impression first.